web/ui/react-app/src/TimeInput.tsx

```diff
@@ -32,19 +32,21 @@ interface TimeInputProps {
 
 class TimeInput extends Component<TimeInputProps> {
   private timeInputRef = React.createRef<HTMLInputElement>();
-  private $time: any | null = null;
+  private $time: any = null;
 
   getBaseTime = (): number => {
     return this.props.time || moment().valueOf();
   };
 
+  calcTimeRange = () => (this.props.range * 1000) / 2;
```

Comment:
```suggestion
  calcShiftRange = () => (this.props.range * 1000) / 2;
```

...might be clearer.